lib/sensu/server/process.rb

```diff
@@ -552,6 +586,7 @@ def determine_stale_clients
             @redis.get("client:#{client_name}") do |client_json|
               unless client_json.nil?
                 client = MultiJson.load(client_json)
+                next if client[:keepalives] == false
```

Comment:
This kinda hints that any JIT client created by an arbitrary check-source will not be auto-pruned (seems legit)
Works for me. This has the effect that every client ever will remain in the client list until manually deleted right? I'm ok with that if you are (not like it hurts anything)